packages/samples/headless-react/src/pages/DependentFacetPage.tsx

```diff
@@ -36,116 +35,241 @@ const Examples = {
 
 type ExampleType = keyof typeof Examples;
 
-export class DependentFacetPage extends Component<
-  {},
-  {currentExample: ExampleType}
-> {
-  private engine: SearchEngine;
-  private readonly authorFacet: HeadlessFacet;
-  private readonly sourceFacet: HeadlessFacet;
-  private readonly objectTypeFacet: HeadlessFacet;
-  private readonly fileTypeFacet: HeadlessFacet;
+type FacetWithDependenciesControllerOptions = FacetOptions &
+  Required<Pick<FacetOptions, 'facetId'>>;
 
-  constructor(props: {}) {
-    super(props);
+interface FacetWithDependenciesProps {
+  engine: SearchEngine;
+  controllerOptions: FacetWithDependenciesControllerOptions;
+  conditions?: AnyFacetValuesCondition<FacetValueRequest>[];
+}
 
-    this.engine = buildSearchEngine({
-      configuration: getSampleSearchEngineConfiguration(),
-    });
+const FacetWithDependencies: FunctionComponent<FacetWithDependenciesProps> = ({
+  engine,
+  controllerOptions,
+  conditions,
+}) => {
```

Comment:
The old doc approach was subscribing to the controller from outside the component and hiding it. The new approach is more composed, it initializes the controller along with its dependencies at the same place where it gets rendered. We don't need multiple subscribers. I prefer this, personally.